RecoParticleFlow/PFClusterProducerAlpaka/plugins/LegacyPFClusterProducer.cc

```diff
@@ -0,0 +1,158 @@
+#include "FWCore/Utilities/interface/EDGetToken.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "DataFormats/Common/interface/Handle.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Utilities/interface/EDPutToken.h"
+#include <Eigen/Core>
+#include <Eigen/Dense>
+#include <algorithm>
+#include <array>
+#include <cmath>
+#include <iostream>
+#include <memory>
+#include <string>
+#include <type_traits>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+#include "DataFormats/DetId/interface/DetId.h"
+#include "DataFormats/ParticleFlowReco/interface/PFRecHitHostCollection.h"
+#include "DataFormats/ParticleFlowReco/interface/PFClusterHostCollection.h"
+#include "DataFormats/ParticleFlowReco/interface/PFRecHitFractionHostCollection.h"
+#include "RecoParticleFlow/PFClusterProducer/interface/PFCPositionCalculatorBase.h"
+#include "RecoParticleFlow/PFClusterProducerAlpaka/interface/PFClusterParamsAlpakaESData.h"
+#include "RecoParticleFlow/PFClusterProducerAlpaka/interface/PFClusterParamsAlpakaESRecord.h"
+
+class LegacyPFClusterProducer : public edm::stream::EDProducer<> {
+public:
+  LegacyPFClusterProducer(edm::ParameterSet const& config)
+      : alpakaPfClustersToken(consumes(config.getParameter<edm::InputTag>("src"))),
+        alpakaPfRHFractionToken(consumes(config.getParameter<edm::InputTag>("src"))),
+        InputPFRecHitSoA_Token_{consumes(config.getParameter<edm::InputTag>("PFRecHitsLabelIn"))},
+        pfClusParamsToken(esConsumes(config.getParameter<edm::ESInputTag>("pfClusterParams"))),
+        legacyPfClustersToken(produces()),
+        _recHitsLabel(consumes(config.getParameter<edm::InputTag>("recHitsSource"))) {
+    edm::ConsumesCollector cc = consumesCollector();
+
+    //setup pf cluster builder if requested
+    const edm::ParameterSet& pfcConf = config.getParameterSet("pfClusterBuilder");
+    if (!pfcConf.empty()) {
+      if (pfcConf.exists("positionCalc")) {
+        const edm::ParameterSet& acConf = pfcConf.getParameterSet("positionCalc");
+        const std::string& algoac = acConf.getParameter<std::string>("algoName");
+        _positionCalc = PFCPositionCalculatorFactory::get()->create(algoac, acConf, cc);
+      }
+
+      if (pfcConf.exists("allCellsPositionCalc")) {
+        const edm::ParameterSet& acConf = pfcConf.getParameterSet("allCellsPositionCalc");
+        const std::string& algoac = acConf.getParameter<std::string>("algoName");
+        _allCellsPositionCalc = PFCPositionCalculatorFactory::get()->create(algoac, acConf, cc);
+      }
+    }
+  }
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+    edm::ParameterSetDescription desc;
+    desc.add<edm::InputTag>("src");
+    desc.add<edm::InputTag>("PFRecHitsLabelIn");
+    desc.add<edm::ESInputTag>("pfClusterParams");
+    desc.add<edm::InputTag>("recHitsSource");
+    desc.setAllowAnything();
```

Comment:
```suggestion
```
Please finalize the `fillDescriptions()`.

Reply:
I added changes to be compatible with the `pfClusterBuilder` from the configuration of `particleFlowClusterHBHE`, but I am not sure still if this the best way to do things.